### PR TITLE
fix: Remove infinite allowance condition to be consistent with erc20 (SC-5348)

### PR DESCRIPTION
**Pull request by vbidin**

### CODE REVIEW

File: contracts/RevenueDistributionToken.sol [hunk 1]
```diff
@@ -274,8 +274,6 @@ contract RevenueDistributionToken is IRevenueDistributionToken, ERC20 {
 
         require(callerAllowance >= shares_, "RDT:CALLER_ALLOWANCE");
 
-        if (callerAllowance == type(uint256).max) return;
-
         allowance[owner_][caller_] = callerAllowance - shares_;
     }
```
Comment (lucas-manuel) [inline contracts/RevenueDistributionToken.sol:278]: We should actually just remove this function entirely and use the _approve function

Comment (edag94) [reply contracts/RevenueDistributionToken.sol:278]: we should use decreaseAllowance

Comment (edag94) [reply contracts/RevenueDistributionToken.sol:278]: > we should use decreaseAllowance

Nevermind its external, we should use `_approve(msg.sender, spender_, allowance[msg.sender][spender_] - subtractedAmount_);`

File: contracts/test/RevenueDistributionToken.t.sol [hunk 2]
```diff
@@ -2516,26 +2516,6 @@ contract RedeemFailureTests is RDTTestBase {
         assertEq(rdToken.allowance(address(shareOwner), address(notShareOwner)), 0);
     }
 
-    function test_redeem_callerNotOwner_infiniteApprovalForCaller() public {
```
Comment (edag94) [inline contracts/test/RevenueDistributionToken.t.sol:2519]: We want infinite approval behavior. The ERC20 code needs to be changed so that this passes

**Review by edag94 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**